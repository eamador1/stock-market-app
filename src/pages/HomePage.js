import React from 'react';
import { Link } from 'react-router-dom';
import hke from '../images/hke.png';
import LondonStockExchange from '../images/LondonStockExchange.png';
import nasdaq from '../images/nasdaq.png';
import euronext from '../images/euronext.jpeg';
import '../style/homePage.css';

const HomePage = () => {
  const options = [
    { title: 'EURONEXT', image: euronext, path: '/euronext' },
    { title: 'NASDAQ', image: nasdaq, path: 'nasdaq' },
    { title: 'LONDON STOCK EXCHANGE', image: LondonStockExchange, path: 'london-exchange' },
    { title: 'HONG KONG EXCHANGE & CLEARING', image: hke, path: 'hke' },

  ];

  return (
    <>
      <header>
        <h1>Stock Wizard</h1>
      </header>
      <section>
        <div className="menuContainer">
          {options.map((option) => (
            <Link to={option.path} key={option.title} className="menuItem">
              <div className="imageContainer">
                <img src={option.image} alt={option.title} />
              </div>
              <h5>
                { option.title }
              </h5>
            </Link>
          ))}
        </div>

      </section>
    </>
  );
};

export default HomePage;
