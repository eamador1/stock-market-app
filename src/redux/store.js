import { configureStore } from '@reduxjs/toolkit';
import companiesReducer from './companies/companiesSlice';

const store = configureStore({
  reducer: {
    home: companiesReducer,
  },
});

export default store;
