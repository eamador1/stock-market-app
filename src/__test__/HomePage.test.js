import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import HomePage from '../components/Home';
import Detail from '../components/CompanyDetail';

describe('Renders without crashing', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Euronext should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<HomePage />} />
            <Route path="/homepage/:content" element={<Detail />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
