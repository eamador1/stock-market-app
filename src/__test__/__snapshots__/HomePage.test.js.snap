// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Euronext should render correctly 1`] = `
<section
  className="wrapper"
>
  <div
    className="headerContainer"
  >
    <h3
      className="exchange"
    >
      EURONEXT
    </h3>
  </div>
  <div
    className="search-bar"
  >
    <input
      className="inputName"
      onChange={[Function]}
      placeholder="Company's Symbol ..."
      type="text"
    />
  </div>
  <div
    className="loadingImage"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.png"
    />
    <h2
      className="loading"
    >
       Loading ...
    </h2>
  </div>
  <div
    className="headerCompanies"
  >
    <h2>
      COMPANIES
    </h2>
  </div>
  <div
    className="stockInfoContainer"
  />
</section>
`;

exports[`Renders without crashing should render correctly 1`] = `
<section
  className="wrapper"
>
  <div
    className="headerContainer"
  >
    <h3
      className="exchange"
    >
      EURONEXT
    </h3>
  </div>
  <div
    className="search-bar"
  >
    <input
      className="inputName"
      onChange={[Function]}
      placeholder="Company's Symbol ..."
      type="text"
    />
  </div>
  <div
    className="headerCompanies"
  >
    <h2>
      COMPANIES
    </h2>
  </div>
  <div
    className="stockInfoContainer"
  />
</section>
`;
